src/pages/Categories.jsx

```diff
@@ -3,10 +3,12 @@ import { fetchCategories } from "../utils/api";
 import { useState } from "react";
 import Loading from "../components/Loading";
 import CategoryCard from "../components/CategoryCard";
+import { Link } from "react-router-dom";
 
-const Categories = () => {
+const Categories = ({ setCategoryName }) => {
```

Comment:
You can't handle it through state, or it doesnt work when someone manually goes to the url, as the state isnt set